designs/cilium-configuration-policy.md

```diff
@@ -0,0 +1,119 @@
+
+##### Validation
+
+As we support more plugins, we will add validations to check that the user has specified only one of the CNIConfigs. These validations will be performed first during the create preflight validations and also as a part of the cluster validating webhook.
+
+
+#### Changes required for the "always" policy enforcement mode
+
+Cilium's `always` policy enforcement mode denies all traffic to and from all pods unless rules/network policies supporting the communication are specified. This means that the EKS Anywhere components, which includes the core Kubernetes deployments, EKS Anywhere deployments and Cluster API deployments will not be able to run without specifying the required network policies. So to allow traffic for these deployments, EKS Anywhere will configure [Kubernetes NetworkPolicy](https://kubernetes.io/docs/concepts/services-networking/network-policies/) resource that selects these deployments and allows traffic to and from them.  
+
+Kubernetes NetworkPolicy is a namespaced resource that allows users to configure ingress and egress rules for pods specified using podSelectors. If a NetworkPolicy's pod selector field is left empty, then its rules apply to all pods within the NetworkPolicy's namespace. 
+When the `always` policy enforcement mode is selected, EKS Anywhere will create NetworkPolicy resources in all core EKS Anywhere namespaces, allowing all ingress and egress traffic to all pods within those namespaces. This will ensure the cluster provisioning and management will go on uninterrupted.  
+This is a sample NetworkPolicy to allow all ingress and egress traffic for workloads in a namespace called `test`:
+
+```yaml
+apiVersion: networking.k8s.io/v1
+kind: NetworkPolicy
+metadata:
+  name: allow-all-ingress-egress
+  namespace: test
+spec:
+  podSelector: {}
+  ingress:
+  - {}
+  policyTypes:
+  - Ingress
+  - Egress
+```
+
+The namespaces within which we need to create such policies will depend on:
+* Whether the cluster is a self-managed/management cluster or a workload cluster managed by another cluster.
+* The infrastructure provider being used.
+
+
+For self-managed/management cluster, EKS Anywhere will create NetworkPolicy resources in the following namespaces allowing all ingress/egress traffic by default:
+- kube-system
+- eksa-system
+- All core Cluster API namespaces:
+  + capi-system
+  + capi-kubeadm-bootstrap-system
+  + capi-kubeadm-control-plane-system
+  + etcdadm-bootstrap-provider-system
+  + etcdadm-controller-system
+  + cert-manager
+- Infrastruture provider's namespace (for instance, capd-system OR capv-system)
+
+For a workload cluster managed by another cluster, EKS Anywhere will create NetworkPolicy resource only in the following namespace by default:
+- kube-system
```

Comment:
Are you thinking of opening ingress to the kube-system pods only to other kube-system pods by default, or all pods across all namespaces?

Reply:
I think ingress from only `kube-system` pods to other `kube-system` pods should suffice on the workload cluster. On the management cluster though, we should be allowing from/to across all the listed namespaces.

Reply:
>On the management cluster though, we should be allowing from/to across all the listed namespaces.

Agreed. But I was thinking the same for workload clusters, since the user apps running on workload clusters would also communicate with kube-apiserver from kube-system. Or should we only allow ingress egress for kube-apiserver across all namespaces on both clusters?

Reply:
I'd rather have networking open only within the `kube-system` namespace for workload cluster. If a user is explicitly selecting the non-default `always` policyEnforcementMode they should expect that they need to open additional networking for their workload pods. With `always` mode, eks-a should only open up networking for cluster ops.

Reply:
modified the doc to show which networkPolicy we'd use in that case. If we find anything wrong with this approach during implementation I'll update the design doc again